### PR TITLE
[RW-8175][risk=no] E2E nightly egress test

### PR DESCRIPTION
**Pull request by aweng98**
Propose some changes hopefully to make the test run reliably.

Screen Shot 2022-04-14 at 1 14 07 PM

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: e2e/utils/runtime-utils.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export async function initializeRuntimeIfModalPresented(page: Page): Promise<voi
 }
 
 async function isSecuritySuspended(page: Page): Promise<boolean> {
-  return exists(page, '//*[@data-test-id="security-suspended-msg"]');
+  return exists(page, '//*[@data-test-id="security-suspended-msg"]', { timeout: 5000 });
```
Comment (aweng98) [inline e2e/utils/runtime-utils.ts:23]: Increase timeout to 5 seconds is needed. The `goto()` triggers page loading which takes few seconds so the default 1 second wait in `exists()` is not enough.

File: e2e/app/page/notebook-page.ts [hunk 2]
```diff
@@ -558,9 +558,6 @@ export default class NotebookPage extends NotebookFrame {
     const fileSizeElement = await newPage.waitForXPath(fileSizeXpath, { visible: true });
     const fileSize = await getPropValue(fileSizeElement, 'textContent');
 
-    // In case page has to be checked after finish.
-    await takeScreenshot(newPage, `notebook-upload-file-${fileName}`);
-
```
Comment (aweng98) [inline e2e/app/page/notebook-page.ts:563]: Don't think it's useful anymore

File: e2e/tests/nightly/egress-suspension.spec.ts [hunk 3]
```diff
@@ -1,33 +1,45 @@
...
+    await (await page.target().createCDPSession()).send('Page.setDownloadBehavior', {
+      behavior: 'allow',
+      downloadPath
+    });
+    await fs.ensureDir(downloadPath);
+  });
 
   test("VM egress suspends user's compute", async () => {
-    await findOrCreateWorkspace(page);
+    await createWorkspace(page);
```
Comment (aweng98) [inline e2e/tests/nightly/egress-suspension.spec.ts:32]: Always create a new workspace

File: e2e/tests/nightly/egress-suspension.spec.ts [hunk 4]
```diff
@@ -36,17 +41,26 @@ describe('egress suspension', () => {
...
-    console.log('Awaiting security suspension in a new page');
-    const newPage = await browser.newPage();
-    await signInWithAccessToken(newPage, config.EGRESS_TEST_USER);
-    // Open the current notebook URL again; this will instead show a suspended
-    // error on reload once the user becomes suspended.
-    await newPage.goto(page.url());
+    await waitForSecuritySuspendedStatus(page);
+  });
+
+  test('Cloud Analysis Environment is auto suspended', async () => {
```
Comment (calbach) [inline e2e/tests/nightly/egress-suspension.spec.ts:57]: What is the purpose of this test case beyond what the original test is doing? In general I'd prefer not to have two interdependent tests like this

**Review by calbach — COMMENTED** [review]
thanks

File: e2e/tests/nightly/egress-suspension.spec.ts (hunk 4 above)
Comment (calbach) [reply e2e/tests/nightly/egress-suspension.spec.ts:57]: The lock status check is fine, but maybe just add this to the end of the other test. I don't think it's worth the complexity of a second dependent test case for the refresh case you describe. The suspension window is not that long (10m), so I really don't want tests that depend on the user being in this state for a prolonged time.

Comment (aweng98) [reply e2e/tests/nightly/egress-suspension.spec.ts:57]: Got it. It's done. Updated PR description with a new screenshot as well. Thank you.

**Review by calbach — APPROVED** [review]
Thanks!

**Pull request merged**